packages/renderer/server/main.ts

```diff
@@ -47,7 +60,38 @@ export const renderVideo = async (
     });
   });
 
-  await page.goto(`http://localhost:${port}/render`);
+  await page.goto(url);
+
+  return renderingComplete;
+}
+
+interface RenderVideoSettings {
+  // Name of the video file
+  name: string;
+}
+
+export const renderVideo = async (
+  configFile: string,
+  params?: Record<string, unknown>,
+  settings: RenderVideoSettings = {
+    name: 'project2',
+  },
+) => {
+  console.log('Rendering...');
+
+  const resolvedConfigPath = path.resolve(process.cwd(), configFile);
+
+  /**
+   * TODO: Change this in the future after more testing.
+   * We still need to concate the videos together after rendering.
+   */
+  const numOfWorkers = 1;
+
+  const promises = [];
+  for (let i = 0; i < numOfWorkers; i++) {
+    const url = buildUrl(9000 + i, settings.name, i, numOfWorkers);
+    promises.push(renderWorker(9000 + i, resolvedConfigPath, url, params));
```

Comment:
I think an error would occur here when port 9000 is already taken - the result of buildUrl would use 9000 but vite uses the next open port by default. IMO, this is okay to leave for now until we add parallel rendering

Reply:
True. Will leave it in and keep it in mind and fix it when we enable multiple workers.